exercise/staff_views.py

```diff
@@ -264,9 +264,9 @@ def post(self, request, *args, **kwargs):
             submitter=self.submission.submitters.first(),
             defaults={'extra_submissions': 0}
         )
-        MaxSubmissionsRuleDeviation.objects\
-            .filter(id=deviation.id)\
-            .update(extra_submissions=(F('extra_submissions') + 1))
+        deviation.extra_submissions += 1
+        deviation.granter = request.user.userprofile
+        deviation.save()
```

Comment:
`QuerySet.update` does not update the `auto_now` fields of models.

Reply:
The advantage of `update` with the `F` field reference is that it is more robust in case multiple requests update the value simultaneously because the old value is used directly in the database. Using `+=` and `save()` fetches the old value into Python, but the value may be changed in the database by another request before the +1 value is written in this request from the Python code. Then, some updates are lost. But we can ignore this since it is not likely going to cause any issues.